### PR TITLE
Add CONOPS Draft

### PR DESCRIPTION
**Pull request by amsmith-pro**
Getting out an initial draft.

Still need to add citations and elaborate on some of the technical and organizational impacts

Too many San Diego examples, over this release I want to accumulate more examples from other cities, including cities that get transit correct.

### CODE REVIEW

File: docs/conops.md [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+Item | Value
+--- | ---
+Maintainer(s) | @arrow-air/services
+
+:exclamation: This document is intended for internal use.
+
+## 1.0 Project Description
+
+#### 1.0.1 Background
+
```
Comment (GoodluckH) [inline docs/conops.md]: Should prob call this "Motivation"

File: docs/conops.md [hunk 2]
```diff
@@ -0,0 +1,549 @@
+# Concept of Operations - Software Services
+
+<center>
```
Comment (GoodluckH) [inline docs/conops.md:3]: Consider adding a TOC.

File: docs/conops.md [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+<center> 
+
+```mermaid
+graph TB
+
+subgraph Provider of Services
+	sched[svc-scheduler]
+	dcvr[svc-discovery]
+	com[svc-compliance]
+	guid[svc-guidance]
```
Comment (GoodluckH) [inline docs/conops.md]: Is there any way to add a blank line between the title and the diagrams? Right now, visually, the title is too close to those sub elements.

File: docs/conops.md [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+
+**Supplemental Data and Services Provider (SDSP)** elements provide *supplemental* or non-critical services for UAM operations.
+
+> "Supplemental Data and Services Provider (SDSP): UAM Operators and PSUs use Supplemental Data Service Providers (SDSPs) to access supporting data including, but not limited to, terrain, obstacle, aerodrome availability, and weather. SDSPs may be accessed via the PSU network or directly by UAM operators." - [UAM Airspace Research Roadmap](https://ntrs.nasa.gov/citations/20210019876)
+
+**Rideshare and Cargo Services** elements are client-facing and not safety-critical.
+
```
Comment (GoodluckH) [inline docs/conops.md]: I'd be careful with the wording "not safety-critical" here for marketing and legal considerations.

File: docs/conops.md [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+`svc-telemetry` | Receives and stores live vehicle telemetry<sup>*</sup><br>Rebroadcasts to ground control/air traffic control tools and other listeners
+`svc-upkeep` | Monitor expiration of vehicle and component certifications<br>Preemptively schedule flights for maintenance
+
+</center>
+
+<sup>*</sup> May qualify as a PSU element.
+#### Rideshare and Cargo Services
+
+These client-facing services expose an API. These are not safety-critical.
+
```
Comment (GoodluckH) [inline docs/conops.md]: ibid re the safety language.

File: docs/conops.md [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+<center> 
+
+Service | Responsibilities
+--- | ---
+`svc-rideshare` | The public API for passengers to request rides through Arrow Services 
+`svc-cargo` | The public API for clients to request cargo transport through Arrow Services
+`svc-charter`| The public API for individuals filing charters for specific aircraft.
+`svc-payment` | Payment processing, accepting traditional payments and cryptocurrencies.
+`svc-contact` | In all other parts of the system, a customer is an integer ID<br>This service has sole access to customer information, the interface to issue notifications to a client
```
Comment (GoodluckH) [inline docs/conops.md]: I thought we're going to use string for UID?

File: docs/conops.md [hunk 7]
```diff
@@ -0,0 +1,549 @@
...
+</center> 
+
+`svc-telemetry`:
+- UAM vehicles broadcast telemetry, which is processed and stored by this service.
+- This service may re-broadcast telemetry to clients such as Air Traffic Control (ATC) software.
+
+`svc-storage`:
+- Links to multiple storage solutions, paired with specific types of data.
+- Blockchain storage provides defensive "immutable" record keeping for certifications, maintenance records, previous flight plans, FAA confirmations.
+- SQL databases for CRUD operations on flight plans
```
Comment (GoodluckH) [inline docs/conops.md]: Have we decided to go forward with SQL? I think we should do some research on what types of db is suitable for flight plans.

File: docs/conops.md [hunk 8]
```diff
@@ -0,0 +1,549 @@
...
+### 3.5 Modes of Operations
+
+Modes of operation correspond with a general fleet routing behavior. They indicate the current goal of flight operations.
+
+<center> 
+
+Mode | Description
+--- | ---
+Nominal | New flight plans may be created if they can be accommodated.<br>Optimize rideshare flights to maximize passengers per aircraft.<br>Optimize flight graph to accommodate more flights per hour.<br>A buzzing flight graph maximizing motion.
+Wind Down | No new flights plans are accepted (save for municipal and emergency flights) for a given time frame.<br>Confirmed flight plans execute as previously planned.<br>Can still add new rideshare passengers to existing flight plans.<br>May occur daily to close flight operations if services are not 24/7.<br>"Chokes" a flight graph to reduce the level of activity.
```
Comment (GoodluckH) [inline docs/conops.md]: Maybe "throttle" is a better word than "choke"?

**Review by GoodluckH — CHANGES_REQUESTED** [review]
This is great! Have a few comments, and I think it's pertinent to have deetwo to redline some legal terms in the doc as well, especially a definition of "Arrow".

File: docs/conops.md (hunk 4 above)
Comment (amsmith-pro) [reply docs/conops.md]: Ah I totally see what you mean. I'll rephrase it to something like "not essential" instead

File: docs/conops.md (hunk 6 above)
Comment (amsmith-pro) [reply docs/conops.md]: Will update, this might be too much specificity anyway

File: docs/conops.md (hunk 7 above)
Comment (amsmith-pro) [reply docs/conops.md]: Will remove 'SQL', too specific anyway

File: docs/conops.md (hunk 8 above)
Comment (amsmith-pro) [reply docs/conops.md]: Absolutely

File: docs/conops.md (hunk 1 above)
Comment (amsmith-pro) [reply docs/conops.md]: Can do

File: docs/conops.md (hunk 3 above)
Comment (amsmith-pro) [reply docs/conops.md]: I can probably add a `<br>` on the subgraph line

File: docs/conops.md (hunk 8 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/fb227c5a17a327041869503558fcb729308c9955

File: docs/conops.md (hunk 7 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/fb227c5a17a327041869503558fcb729308c9955

File: docs/conops.md (hunk 6 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/fb227c5a17a327041869503558fcb729308c9955

File: docs/conops.md (hunk 5 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/fb227c5a17a327041869503558fcb729308c9955

File: docs/conops.md (hunk 4 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/fb227c5a17a327041869503558fcb729308c9955

File: docs/conops.md (hunk 2 above)
Comment (amsmith-pro) [reply docs/conops.md:3]: https://github.com/Arrow-air/se-services/pull/2/commits/8dae069f47dbe1702800a35f75bdd00db8a117c3

Used `auto-markdown-toc` extension

File: docs/conops.md (hunk 3 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/4b0f31ca0d9e2d1a9436e5b4ff771622dfa5ce9e

Padding with mermaid is sketchy. The best solution was to wrap the contents within another transparent subgraph

File: docs/conops.md (hunk 1 above)
Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/4b0f31ca0d9e2d1a9436e5b4ff771622dfa5ce9e

File: docs/conops.md [hunk 9]
```diff
@@ -12,9 +12,70 @@ Maintainer(s) | @arrow-air/services
 
 :exclamation: This document is intended for internal use.
 
+## Table of Contents
+
+<!-- TOC -->
+
```
Comment (GoodluckH) [inline docs/conops.md]: Many headings don't seem to be linking.

**Review by GoodluckH — CHANGES_REQUESTED** [review]

File: docs/conops.md (hunk 9 above)
Comment (amsmith-pro) [reply docs/conops.md]: Agreed with reviewer to remove manual TOC and rely on:
- GitHub Autogenerated TOC: https://github.blog/changelog/2021-04-13-table-of-contents-support-in-markdown-files/
- Autogenerated Docusaurus TOC

Comment (amsmith-pro) [reply docs/conops.md]: https://github.com/Arrow-air/se-services/pull/2/commits/e0f2f98ad6a6085054ba4c248e7a176c73f8ac7c

**Review by GoodluckH — DISMISSED** [review]

File: docs/conops.md [hunk 10]
```diff
@@ -0,0 +1,582 @@
...
+subgraph Arrow Rideshare & Cargo Services
+classDef subgraph_padding fill:none,stroke:none
+
+    subgraph p [ ]
+	share[svc-rideshare]
+	cargo[svc-cargo]
+    charter[svc-charter]
+	pay[svc-payment]
+	contact[svc-contact]
+    end
```
Comment (owlot) [inline docs/conops.md:84]: We also have svc-pricing now (-;

File: docs/conops.md [hunk 11]
```diff
@@ -0,0 +1,582 @@
...
+<center> 
+
+Service | Responsibilities
+--- | ---
+`svc-rideshare` | The public API for passengers to request rides through Arrow Services 
+`svc-cargo` | The public API for clients to request cargo transport through Arrow Services
+`svc-charter`| The public API for individuals filing charters for specific aircraft.
+`svc-payment` | Payment processing, accepting traditional payments and cryptocurrencies.
+`svc-contact` | This service has sole access to customer information, the interface to issue notifications to a client
+
```
Comment (owlot) [inline docs/conops.md:173]: add `svc-pricing` as the service used to request pricing quotes during flight planning

File: docs/conops.md (hunk 10 above)
Comment (amsmith-pro) [reply docs/conops.md:84]: https://github.com/Arrow-air/se-services/pull/2/commits/fd87e1512c5648b706fd9161d66063f763adc043

File: docs/conops.md (hunk 11 above)
Comment (amsmith-pro) [reply docs/conops.md:173]: https://github.com/Arrow-air/se-services/pull/2/commits/fd87e1512c5648b706fd9161d66063f763adc043

**Review by GoodluckH — DISMISSED** [review]

**Review by GoodluckH — DISMISSED** [review]

**Review by owlot — APPROVED** [review]

**Review by GoodluckH — APPROVED** [review]
